From 4a806eac55b079ed054d85938c82705459c3cd92 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Thu, 22 Jun 2017 19:40:56 +0200
Subject: [PATCH 5/6] SensitivePhoneNumbers: Handle lists of MCC codes

* Some countries have multiple MCC codes, so handle it
* In the sensitivePN network field, enter the list of
  MCCs like so: https://review.lineageos.org/178233

Change-Id: I34225473404b2be2640ea9ab05691dc985c49fa0
---
 src/com/android/server/telecom/SensitivePhoneNumbers.java | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/com/android/server/telecom/SensitivePhoneNumbers.java b/src/com/android/server/telecom/SensitivePhoneNumbers.java
index e652bc04..b5703b2c 100644
--- a/src/com/android/server/telecom/SensitivePhoneNumbers.java
+++ b/src/com/android/server/telecom/SensitivePhoneNumbers.java
@@ -86,8 +86,11 @@ public class SensitivePhoneNumbers {
             if ("sensitivePN".equals(name)) {
                 SensitivePhoneNumber sensitivePN = SensitivePhoneNumber
                         .readSensitivePhoneNumbers(parser);
-                mSensitiveNumbersMap.put(sensitivePN.getNetworkNumeric(),
-                        sensitivePN.getPhoneNumbers());
+                String[] mccs = sensitivePN.getNetworkNumeric().split(",");
+                ArrayList<String> sensitive_nums = sensitivePN.getPhoneNumbers();
+                for (String mcc : mccs) {
+                    mSensitiveNumbersMap.put(mcc, sensitive_nums);
+                }
             } else {
                 break;
             }
-- 
2.17.1

